### DISCUSSION TITLE
An error was encountered while creating the dataset

### DISCUSSION THREAD

**Discussion by yaoysyao** [pyg-team/pytorch_geometric#7213] [Q&A]
I am currently doing text classification, using graph classification method. Now I need to construct data set conforming to PYG format according to data set. I refer to the TUDataset class to construct data set, because this data set is also used for graph classification. And validate(raise_on_error=True) method to prove my data is correct, I refer to the source code found that the TUDataset method calls the split(data, batch) method.
I would like to ask you the following questions:
1. What does this method do?
2. I also called this method, but found that edge_index was negative, resulting in the following error :RuntimeError: index -14102 is out of bounds for dimension 0 with size 8817,  The problem occurs in data.edge_index -= node_slice[batch[row].unsqueeze(0)
3. The diagram constructed by TUDataset class is how to assemble nodes and graphs. I see that there are nodes in the data set belonging to a txt file of which graph, but how to judge the belonging of each node in the code? How do you determine which graph each node belongs to?

**Comment by rusty1s** [comment]
Hi, I am not sure if `io/tu.py` is absolutely necessary for you to look into as it contains parsing logic to convert the raw graph format defined in https://chrsmrrs.github.io/datasets/ to PyG format. Regarding your questions:
1/2: The graph format of `TUDataset` stores all graphs in a single file, so `split` takes care of splitting this dataset into individual components. As such, each `edge_index` part is decreased to start from index zero (this is the `data.edge_index -= node_slice[batch[row].unsqueeze(0)` line).
3: There exists a `DS_graph_indicator.txt` text file which contains this information.

**Reply by yaoysyao** [reply]
Thank you very much for your answer,  I am currently working on text categorizing. I have looked at the examples in the source code, but I don't seem to find any examples of text categorizing, such as TextGcn and TextlevelGnn. However, due to the failure to correctly construct data format conforming to PYG, I encountered some problems. Is there any plan to add relevant code in the example, which will be very helpful to me

**Reply by rusty1s** [reply]
Can you do me a favor and create an issue for the models you would like to see in PyG? That would be very helpful.
